rcljava/src/main/java/org/ros2/rcljava/graph/NameAndTypes.java

```diff
@@ -0,0 +1,92 @@
+/* Copyright 2020 Open Source Robotics Foundation, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ros2.rcljava.graph;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Objects;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.ros2.rcljava.common.JNIUtils;
+
+/**
+ * Class that represents a topic/service/action name, together with all the types
+ * of message/service/action that are using that name.
+ */
+public class NameAndTypes {
+  /// Name of the topic/service/action.
+  public String name;
+  /// Types of the topic/service/action with the given name.
+  public Collection<String> types;
+
+  /**
+   * Construct from given name and types.
+   *
+   * @param name name of the topic/service/action.
+   * @param types types of the given topic/service/action.
+   *  A shallow copy of the given collection will be stored,
+   *  but given that String is immutable, this is not a problem.
+   * @param typesSize size of the \a typesHandle array.
+   */
+  public NameAndTypes(final String name, final Collection<String> types) {
+    this.name = name;
+    this.types = new ArrayList(types);
+  }
+
+  /// @internal Default constructor, only used from jni code.
+  private NameAndTypes() {
+    this.types = new ArrayList();
+  }
+
+  /**
+   * @internal Create a NameAndTypes object from jni code.
+   *
+   * @param nameHandle pointer to a C string.
+   * @param typesHandle pointer to an array of C strings.
+   * @param typesSize size of the \a typesHandle array.
+   */
+  private native final void nativeFromRCL(long nameHandle, long typesHandle, long typesSize);
```

Comment:
Is this native method necessary? It doesn't look like it is used anywhere and it doesn't correspond to an rcl type.

I think creating the `NamesAndTypes` objects from `NodeImpl.nativeGetTopicNamesAndTypes` is sufficient.

Reply:
done